flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -208,6 +219,8 @@ public void shutDown() throws FlinkException {
 
 		taskEventDispatcher.clearAll();
 
+		taskManagerMetricGroup.close();
```

Comment:
if we decide to keep `taskManagerMetricGroup` in `TaskManagerServices` then `taskManagerMetricGroup.close();` is not needed in `TaskExecutor.stopTaskExecutorServices` and probably as a field there at all?

Reply:
That is true. We decide not to make `TaskManagerMetricGroup` as one component of `TaskManagerServices`, so keep the previous behavior in `TaskExecutor`.